### DISCUSSION TITLE
TAO Exceptions based on std::exception

### DISCUSSION THREAD

**Discussion by tstaig** [DOCGroup/ACE_TAO#2007] [Q&A] [answered]
I was discussing with some people at our project and we noticed that the latest official C++ mapping document [1] states that the Exception class (defined in TAO/tao/Exception.h) should extend the std::exception class.

So I wanted to ask why haven't ACE/TAO adopted this mapping standard? Is it just because there are more pressing matters? Or there are consequences such as marshalling/unmarshalling issues o performance hit?

Thanks!

[1]: https://www.omg.org/spec/CPP11/1.1/PDF/  (Page 35, section 6.20)

**Comment by jwillemsen** [comment, marked as answer]
You refer to the IDL to C++11 language mapping (see https://www.omg.org/spec/CPP11) which is supported by TAOX11, see https://www.taox11.org. TAOX11 is an extension of TAO which implements this new mapping, TAO itself uses the IDL to C++ mapping, see https://www.omg.org/spec/CPP

**Reply by tstaig** [reply]
Thanks @jwillemsen for your answer, I was unaware there was a separate TAO project for C++11 mappings. I presume there will be several other changes that could impact a migration, but we will study this possibility.

**Reply by jwillemsen** [reply]
The IDL to C++11 language mapping is much easier and safer to use, but you have to migrate your code, mostly a find/replace effort, the compiler will tell you where you have to change things.
